main.cpp

```diff
@@ -1,37 +1,74 @@
 #include <iostream>
 #include <fstream>
 #include <string>
+#include <sstream>
+#include <cassert>
 
-bool test_mode = true;
+namespace colors
+{
+    char red[] = "\033[31m";
+    char blue[] = "\033[34m";
+    char none[] = "\033[0m";
+}
+
+bool test_mode = false;
 
-int main()
+//requires minimum 3 arguments
+//maximum 4
+int main(int argc, char *argv[])
 {
-    std::string file_in_path = "./in.f", file_out_path = "./out.f";
-    /*
-    std::cout << "Name of in file:";
-    std::cin >> file_in_path;
-    std::cin.clear();
-    std::cin.ignore(std::numeric_limits<std::streamsize>::max(),'\n');
-    */
+    if(argc > 4 || argc < 3)
+    {
+        std::cout << colors::red << "Got " << argc << " arguments, but 3 or 4 needed" <<
+        std::endl << "Terminating" << colors::none << std::endl;
+        return 1;
+    }
+    std::string file_in_path = argv[1], file_out_path = argv[2];
+
+    if(file_in_path == file_out_path)
+    {
+        std::cout << colors::red << "AAAA Fuck!" << colors::none << std::endl;
```

Comment:
1. Please use more clear error messages.
2. Read about the difference btw std::endl and "\n". Decide which one is better to use in this case (and further in the code)